### PR TITLE
feat: implement abstraction level for feedback service

### PR DESCRIPTION
**Pull request by Inerska**
This PR implements an higher abstraction level for FeedbackService, it will allow people to mock FeedbackService easier.

### CODE REVIEW

File: Remora.Discord.Commands/Feedback/Services/FeedbackService.cs
Comment (VelvetThePanda) [inline Remora.Discord.Commands/Feedback/Services/FeedbackService.cs:1]: Documentation on this service should be changed to `/// <inheritdoc>`

**Review by VelvetThePanda — COMMENTED** [review]
LGTM

File: Remora.Discord.Commands/Feedback/Services/FeedbackService.cs
Comment (Inerska) [reply Remora.Discord.Commands/Feedback/Services/FeedbackService.cs:1]: Fixed with the new commit.

**Comment by mdabrarhassan** [comment]
https://github.com/Remora/Remora.Discord/blob/main/Remora.Discord.Commands/Extensions/ServiceCollectionExtensions.cs#L249

Should the abstraction not also be registered here?

File: Remora.Discord.Commands/Feedback/Services/IFeedbackService.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    ///     Sends a feedback message with the given content and color in the current context, optionally mentioning a target
+    ///     user.
+    /// </summary>
+    /// <param name="contents">The content of the feedback message.</param>
+    /// <param name="color">The color of the feedback message embed.</param>
+    /// <param name="target">The optional target user to mention in the feedback message.</param>
+    /// <param name="options">Optional custom message options.</param>
+    /// <param name="ct">Optional CancellationToken to cancel the operation.</param>
+    /// <returns>A Result containing a list of sent messages, or an error if the operation failed.</returns>
+    Task<Result<IReadOnlyList<IMessage>>> SendContextualContentAsync(
```
Comment (Nihlus) [inline Remora.Discord.Commands/Feedback/Services/IFeedbackService.cs:49]: Formatting. The opening brace should be placed on a new line if you can't fit it all on the same line, same way you'd handle { and }.

Check the other methods for this as well.

File: Remora.Discord.Commands/Feedback/Services/IFeedbackService.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+using System.Threading.Tasks;
+using Remora.Discord.API.Abstractions.Objects;
+using Remora.Discord.Commands.Feedback.Messages;
+using Remora.Rest.Core;
+using Remora.Results;
+
+namespace Remora.Discord.Commands.Feedback.Services;
+
+/// <summary>
+///     Handles sending formatted messages to the users.
```
Comment (Nihlus) [inline Remora.Discord.Commands/Feedback/Services/IFeedbackService.cs:35]: Do not indent text within documentation comments.

Check the other documentation comments as well.

File: Remora.Discord.Commands/Feedback/Services/FeedbackService.cs [hunk 3]
```diff
@@ -39,11 +39,10 @@
 
 namespace Remora.Discord.Commands.Feedback.Services;
 
-/// <summary>
-/// Handles sending formatted messages to the users.
-/// </summary>
+/// <inheritdoc />
 [PublicAPI]
 public class FeedbackService
+    : IFeedbackService
```
Comment (Nihlus) [inline Remora.Discord.Commands/Feedback/Services/FeedbackService.cs:45]: This should be on the same line as the class declaration unless required.

File: Remora.Discord.Commands/Feedback/Services/FeedbackService.cs [hunk 4]
```diff
@@ -39,11 +39,10 @@
 
 namespace Remora.Discord.Commands.Feedback.Services;
 
-/// <summary>
-/// Handles sending formatted messages to the users.
-/// </summary>
+/// <inheritdoc />
```
Comment (Nihlus) [inline Remora.Discord.Commands/Feedback/Services/FeedbackService.cs:42]: Replace the documentation comments on the interface-implemented methods with inheritdoc as well.

File: Remora.Discord.Commands/Feedback/Services/IFeedbackService.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    Task<Result<IMessage>> SendPrivateAsync(
+        Snowflake user,
+        Optional<string> content = default,
+        Optional<IReadOnlyList<IEmbed>> embeds = default,
+        FeedbackMessageOptions? options = null,
+        CancellationToken ct = default
+    );
+
+    /// <summary>
+    ///     Checks if the current context is an InteractionContext.
```
Comment (Nihlus) [inline Remora.Discord.Commands/Feedback/Services/IFeedbackService.cs:124]: Reuse the existing documentation on FeedbackService where possible. This goes for all the methods.

**Review by Nihlus — CHANGES_REQUESTED** [review]
Looks good, mostly issues with formatting :)

I would like you to modify the service registration as well so that end users can request the interface instead of the concrete type. The samples should probably be adjusted to do so as well.

**Comment by MazeXP** [comment]
> https://github.com/Remora/Remora.Discord/blob/main/Remora.Discord.Commands/Extensions/ServiceCollectionExtensions.cs#L249
> 
> Should the abstraction not also be registered here?

You are totally right on this one!

File: Remora.Discord.Commands/Extensions/ServiceCollectionExtensions.cs [hunk 6]
```diff
@@ -246,7 +246,7 @@ public static IServiceCollection AddDiscordCommands
 
         serviceCollection.TryAddSingleton<ExecutionEventCollectorService>();
 
-        serviceCollection.TryAddScoped<FeedbackService>();
+        serviceCollection.TryAddScoped<IFeedbackService, FeedbackService>();
```
Comment (MazeXP) [inline Remora.Discord.Commands/Extensions/ServiceCollectionExtensions.cs:249]:
```suggestion
        serviceCollection.TryAddScoped<FeedbackService>();
        serviceCollection.TryAddScoped<IFeedbackService>(s => s.GetRequiredService<FeedbackService>());
```
To keep backwards compatibility.

**Review by MazeXP — CHANGES_REQUESTED** [review]

**Review by Nihlus — APPROVED** [review]

**Pull request merged**